packages/TorneloScoresheet/src/hooks/appMode/editMoveState.ts

```diff
@@ -96,15 +97,22 @@ const rebuildHistory = (
   }
   let previousPly: ChessPly = editedPly;
   let previousStartingFen = getStartingFen(previousPly) ?? '';
-  if (previousStartingFen === '') {
-    return null;
-  }
 
   // for each move after the edited ply
   // we generate it's new starting fen based on the previous ply
   // if the starting fen is null (the move was impossible) we add null to to newHistory[]
   // otherwise we add the ply with the updated starting fen to newHistory[]
   const newHistory = futureMoves.map(move => {
+    if (previousStartingFen === '') {
+      previousPly = {
+        moveNo: move.moveNo,
+        startingFen: previousPly.startingFen,
+        drawOffer: move.drawOffer,
+        player: move.player,
+        type: PlyTypes.SkipPly,
```

Comment:
Missing the legality check here, it should be performed with the fen after this skip ply is recorded